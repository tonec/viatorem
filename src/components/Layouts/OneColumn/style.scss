
.row {
  display: flex;
  height: 100%;
  align-items: flex-start;
}