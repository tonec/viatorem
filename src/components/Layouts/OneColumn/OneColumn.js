import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import classNames from 'classnames'
import { Layout, Row, Col } from 'antd'
import { Navigation } from 'containers'

import style from './style.scss'

const { Header, Content } = Layout

const propTypes = {
  span: PropTypes.number,
  offset: PropTypes.number,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

const defaultProps = {
  span: 24,
  offset: 0,
  className: ''
}

const OneColumn = ({ span, offset, title, className, children, ...props }) => {
  const classes = classNames('layout one-column', className)

  return (
    <div className={classes} {...props}>
      <Helmet title={title} />
      <Header className="layout-header">
        <Navigation />
      </Header>
      <Content className="layout-content">
        <Row className={style.row}>
          <Col span={span} offset={offset}>
            {children}
          </Col>
        </Row>
      </Content>
    </div>
  )
}

OneColumn.propTypes = propTypes
OneColumn.defaultProps = defaultProps

export default OneColumn
