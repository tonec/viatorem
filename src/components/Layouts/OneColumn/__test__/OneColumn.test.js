import React from 'react'
import { shallow } from 'enzyme'
import Helmet from 'react-helmet'
import OneColumn from '../OneColumn'

const props = {
  title: 'Test title',
  className: 'test-class-name'
}

const childContent = <p>Child content</p>

describe('OneColumn (snapshot)', () => {
  it('renders according to snapshot when enabled', () => {
    const component = shallow(<OneColumn {...props}>{childContent}</OneColumn>)
    expect(component).toMatchSnapshot()
  })
})

describe('OneColumn (rendering)', () => {
  let component

  beforeEach(() => {
    component = shallow(<OneColumn {...props}>{childContent}</OneColumn>)
  })

  it('renders the correct title', () => {
    expect(component.find(Helmet)).toHaveLength(1)
    expect(component.find(Helmet).props().title).toBe(props.title)
  })

  it('includes the correct class name', () => {
    expect(component.find('.one-column')).toHaveLength(1)
  })
})
