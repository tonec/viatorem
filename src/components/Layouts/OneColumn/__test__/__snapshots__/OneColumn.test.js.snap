// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneColumn (snapshot) renders according to snapshot 1`] = `
<div
  className="layout one-column test-class-name"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Test title"
  />
  <Adapter
    className="layout-header"
  >
    <Connect(Navigation) />
  </Adapter>
  <Adapter
    className="layout-content"
  >
    <Row
      gutter={0}
    >
      <Col
        offset={0}
        span={24}
      >
        <p>
          Child content
        </p>
      </Col>
    </Row>
  </Adapter>
</div>
`;
